doc/L2_802.1q_Tunneling_Support_HLD/L2 dot1q HLD.markdown

```diff
@@ -0,0 +1,460 @@
+- Feature Design  
+  - Sample Topology  
+  - Flow Chart  
+    - Flow Chart Explanation  
+  - Architecture Design  
+- High-level Design  
+  - Part 1  
+  - Part 2  
+  - DB & Schema Changes  
+  - Warm Reboot Requirements  
+  - Fast reboot Requirements  
+- SAI API  
+  - Configuration and Management  
+  - VLAN DOT1Q Configuration Commands  
+  - VLAN DOT1Q Show Commands  
+- Warm Boot and Fastboot Design Impact  
+- Restrictions/Limitations  
+  - Testing Requirements/Design  
+    - Unit Test cases  
+
+## Scope
+
+This high-level design document describes the integration of VLAN tunneling encapsulation protocol IEEE 802.1Q in SONiC.  
+
+## Definitions/Abbreviations
+
+| **Sr No** |  **Term**   |  **Definition**                      |
+| :------------- |:-------------| :-----|
+|  1       |  VLAN_ID      |  Unique identifier for each VLAN     |  
+|  2       |  Dot1q        |  IEEE 802.1Q Protocol                |  
+|  3       |  Trunk Link   |  The link between different switches that carries the traffic from different VLANs  |  
+|  4       | Encapsulation |  Addition of dot.1q tag in the normal ethernet frame.|  
+|  5       |  Egress Port  |  Outbound port                       |  
+|  6       |  Ingress Port |  Inbound port                        |  
+|  7       |  BGP          |  Border gateway protocol             |  
+|  8       |  DHCP         |  Dynamic Host Configuration Protocol |  
+|  9       |  LAG          |  Link Aggregation                    |  
+|  10      |  VID          |  VLAN Identifier                     |  
+|  11      |  TPID         |  Tag Protocol Identifier             |  
+
+## Overview
+
+L2 DOT1Q (802.1Q) is an IEEE standard for tunnel encapsulation to support transport of different VLAN frames on the tunnel link. This document covers the key aspects of IEEE 802.1Q Tunneling protocol which will be implemented in the existing architecture of the SONiC. 802.1Q support is very fundamental for the forwarding of different VLANs frames on the tunnel links. It is also mentioned in the backlog of the SONiC roadmap that the addition of [L2 IEEE 802.1Q Tunneling Support](https://github.com/Azure/SONiC/wiki/Sonic-Roadmap-Planning) in SONiC is yet to be done. Thus, this HLD is prepared to provide design and implementation details of L2 Dot1Q tunneling support in SONiC.  
+
+## Introduction
+
+IEEE 802.1Q is a networking standard which defines the VLANs tagging on an ethernet network. The Ethernet frame contains a TAG with fields that specify VLAN membership and user priority. As seen in the below Ethernet frame, the TAG is inserted between the Source MAC address and Type/Length. The TAG is of 16-bit length. The 802.1Q tags are inserted into the frames before  transmitting out of the interface and similarly, the tags are removed from frames received from the interface. The frames are transmitted and received from the interfaces that are configured with VLAN and they are tagged with VID.  
```

Comment:
I believe, there is a HLD for VLAN stacking (https://github.com/Azure/SONiC/pull/915/files), this HLD seems duplicate, please clarify.

Reply:
To the best of our understanding, our HLD and HLD 915 (VLAN Stacking), submitted about a week after our HLD, are very different. Our HLD covers the 802.1Q trunking encapsulation implementation within a client network. On the contrary, HLD 915 covers the VLAN staking Q-in-Q which is related to the ISP network. So, there is no duplication in both the HLDs.

Reply:
Per my understanding, the outer tag TPID is 0x8100 instead of 0x88a8 in the HLD 915, and I think, your encapsulation also does the same thing, please read the HLD 915 and combine the HLDs if required.

Reply:
In this HLD we have implemented 802.1Q standard with respect to client networks which performs frame tagging for carrying different VLANs traffic on the trunk link within a client network. Only one tag is added within the client network (No stacking). On the other hand, HLD # 915 has implemented 802.1ad which deals with the forwarding of VLAN IDs (VIDs) between geographically separated different sites of the clients connected via ISPs. ISPs use 802.1ad for stacking VID from different clients and add an ISP tag on top of those stacked VIDs. So, 802.1ad deals with the tagging inside an ISPs network and 802.1Q deals with the tagging inside clients networks. There is another type of tagging according to 802.1ah standard which deals with another layer of stacking of ISP's tags inside Backbone networks. All of these three Vlan tagging standards are independent and have a completely disjoint scope of operation. These standards are identified by switches from different values of TPID field inside the tags, 81-00 for customer VLAN Tag, 88-a8 for Service Provider or Backbone VLAN Tags and 88-e7 for Backbone Service Instance Tag. Please refer to table 9-1 of the IEEE 802.1Q-2018 standards guide for the details of technical differences in these standards. The IEEE 802.1Q-2018 standards guide is available at: https://standards.ieee.org/standard/802_1Q-2018.html. Thus, The scope of this HLD and HLD#915 are very different and are implementing related but different and disjoint standards. In SONiC roadmap "L2 Dot1Q tunneling support" is mentioned so we started implementation from the 802.1Q baseline standard and we have had plans for implementing 802.1ad and 802.1ah in separate HLDs.

Reply:
This feature is not in community release tracking list.